File: 1482-how-many-numbers-are-smaller-than-the-current-number/how-many-numbers-are-smaller-than-the-current-number.py
class Solution:
    def smallerNumbersThanCurrent(self, nums: List[int]) -> List[int]:
        #brute force
        numSorted = sorted(nums)
        res = []
        mapA = {}
        for n in range(0, len(numSorted)):
            if numSorted[n] not in mapA:   
                # we only update for the first occuracnce of an element to mitigate errors due to duplicate numbers
                mapA[numSorted[n]] = n
        for m in nums:
            res.append(mapA[m])
        return res

File: 1482-how-many-numbers-are-smaller-than-the-current-number/how-many-numbers-are-smaller-than-the-current-number.java
class Solution {
    public int[] smallerNumbersThanCurrent(int[] nums) {
        int[] originalNums = Arrays.copyOf(nums, nums.length);
        Arrays.sort(nums);
        HashMap<Integer,Integer> map = new HashMap<>();
        for (int i = 0 ; i < nums.length ; i++){
            if (map.containsKey(nums[i])){
                continue;
            }
            else{
                map.put(nums[i] , i);
            }
        }
        int[] res = new int[nums.length];
        for (int i = 0 ; i < nums.length ; i++){
            res[i] = map.get(originalNums[i]);
        }

        return res;
    }
}